### PR TITLE
[Codechange] Adapted the (UNIX) CXX_FLAGS from rigs-of-rods/CMakeLists.txt

### PR DESCRIPTION
**Pull request by ulteq**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,14 +36,19 @@ if(WIN32)
   set(CMAKE_EXE_LINKER_FLAGS_MINSIZEREL     "${CMAKE_EXE_LINKER_FLAGS_MINSIZEREL}     /DEBUG /LTCG /OPT:REF")
   set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} /DEBUG /LTCG /OPT:REF")
 ELSEIF(UNIX)
-  set(CMAKE_EXE_LINKER_FLAGS_RELEASE        "${CMAKE_EXE_LINKER_FLAGS_RELEASE}      -std=c++11 -O2 -march=nocona -ffast-math")
-  set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} -std=c++11 -Og -g")
-
-  # add gdb debugging stuff
-  set(CMAKE_CXX_FLAGS_DEBUG                 "-std=c++11 -pedantic -Wall -Werror -ggdb ${CMAKE_CXX_FLAGS_DEBUG}")
```
Comment (Hiradur) [inline CMakeLists.txt]: these flags should be kept to keep the code clean in the future.

**Comment by ulteq** [comment]
Like this?

`set(CMAKE_CXX_FLAGS                       "-std=c++11 -pedantic -Wall -Werror ${CMAKE_CXX_FLAGS}")`

**Comment by Hiradur** [comment]
yes, putting them into the general flags looks like a good idea.

**Comment by ulteq** [comment]
> yes, putting them into the general flags looks like a good idea.

Guess we can't do it like that? Travis failed for some reason.

**Comment by Hiradur** [comment]
That's because Travis uses gcc 4.6 which doesn't have full support for C++11. Travis would have to install a newer version similar to how it does for the client.

**Comment by Hiradur** [comment]
:+1:

**Comment by ulteq** [comment]
I will squash some of the commits when I'm at home.

**Pull request merged**